### PR TITLE
Nightly builds prerequisites

### PR DESCRIPTION
**Pull request by ArturGajowy**
Prerequisites for the nightly (or rather: 3-hourly) [presto-checks]
(https://travis-ci.org/Teradata/presto-checks) build.

The `presto-checks` build executes all the product tests, including `big_query`, `storage_formats` and tests for authentication mechanisms (with and without **kerberos**, with and without **impersonation**). We're going to add **LDAP** product tests run in there and maybe run with Hortonworks hadoop distribution as well (we only test with Cloudera so far). For details on what tests are run by `presto-checks`, see [here](https://github.com/Teradata/presto-checks/blob/master/.travis.yml#L41).

The `presto-checks` build reuses artifacts uploaded during the main presto build and runs for [Presto master](https://travis-ci.org/Teradata/presto-checks/builds/158192533) (emulated by branch `nightly_builds_prerequisites` until this PR is merged) and our [sprint branch](https://travis-ci.org/Teradata/presto-checks/builds/158180799).

---
To achieve that, this PR:
 - allows running product tests with arbitrary / downloaded artifacts
 - allows for overriding the image used for `hadoop-master` container in product tests
 - makes the `big_query` tests pass on Travis by increasing `query.max-memory` and `query.max-memory-per-node` in the `singlenode` configuration to 2GB and 1GB respectively.

To achieve the first point above, this PR also reorganizes docker volumes to only mount necessary parts of presto and be more clear about what's needed in product tests

TD internal review here: Teradata#313.

Supersedes #5928.

### CODE REVIEW

**Comment by ArturGajowy** [comment]
@martint please have a look, that's a small change giving big benefits when testing Presto

File: presto-product-tests/bin/locations.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (electrum) [inline presto-product-tests/bin/locations.sh]: Since this is only sourced and not executed directly, it doesn't need a shebang line.

Comment (electrum) [inline presto-product-tests/bin/locations.sh]: Also should not be executable

Comment (ArturGajowy) [inline presto-product-tests/bin/locations.sh]: @electrum fixed, both for locations.sh and compose-commons.sh. Anything else? :)

**Comment by ArturGajowy** [comment]
@electrum @martint I applied your comments. Anything else? :)

**Comment by electrum** [comment]
The commit message "Increase query.max-memory ..." would be better titled "Increase query memory limits for singlenode product tests". There's no need to mention the config variable names in the description -- you can see that from the diff.

**Comment by electrum** [comment]
I'm going to trust the other review for "Reorganize docker volumes" since I have no idea what's going on there.

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**